src/expr.rs

```diff
@@ -961,7 +963,8 @@ impl Rewrite for ast::Arm {
 
             match rewrite {
                 Some(ref body_str) if !body_str.contains('\n') || !context.config.wrap_match_arms ||
-                                      comma.is_empty() => {
+                                      comma.is_empty() ||
```

Comment:
I don't think these are the right conditions. We should check whether the body is a block here. This will currently produce incorrect results for non-body blocks when `match_block_trailing_comma == true`.